Unity/Assets/MyScripts/ShowInterstitialScript.cs

```diff
@@ -25,67 +37,77 @@ void Start ()
 		ShowText = GameObject.Find ("ShowInterstitialText");
 		ShowText.GetComponent<UnityEngine.UI.Text> ().color = UnityEngine.Color.red;
 
-		// Add Interstitial Events
-		IronSourceInterstitialEvents.onAdReadyEvent += InterstitialOnAdReadyEvent;
-		IronSourceInterstitialEvents.onAdLoadFailedEvent += InterstitialOnAdLoadFailed;
-		IronSourceInterstitialEvents.onAdOpenedEvent += InterstitialOnAdOpenedEvent;
-		IronSourceInterstitialEvents.onAdClickedEvent += InterstitialOnAdClickedEvent;
-		IronSourceInterstitialEvents.onAdShowSucceededEvent += InterstitialOnAdShowSucceededEvent;
-		IronSourceInterstitialEvents.onAdShowFailedEvent += InterstitialOnAdShowFailedEvent;
-		IronSourceInterstitialEvents.onAdClosedEvent += InterstitialOnAdClosedEvent;
+		
 
 	}
 
 	/************* Interstitial API *************/ 
 	public void LoadInterstitialButtonClicked ()
 	{
+		// Create interstitial Ad
+		interstitialAd = new LevelPlayInterstitialAd(interstitialAdUnitId);
+		
+		// Register to events
+		interstitialAd.OnAdLoaded += InterstitialOnAdLoadedEvent;
+		interstitialAd.OnAdLoadFailed += InterstitialOnAdLoadFailedEvent;
+		interstitialAd.OnAdDisplayed += InterstitialOnAdDisplayedEvent;
+		interstitialAd.OnAdDisplayFailed += InterstitialOnAdDisplayFailedEvent;
+		interstitialAd.OnAdClicked += InterstitialOnAdClickedEvent;
+		interstitialAd.OnAdClosed += InterstitialOnAdClosedEvent;
+		interstitialAd.OnAdInfoChanged += InterstitialOnAdInfoChangedEvent;
+		
 		Debug.Log ("unity-script: LoadInterstitialButtonClicked");
-		IronSource.Agent.loadInterstitial ();
+		interstitialAd.LoadAd();
 	}
 	
 	public void ShowInterstitialButtonClicked ()
 	{
 		Debug.Log ("unity-script: ShowInterstitialButtonClicked");
-		if (IronSource.Agent.isInterstitialReady ()) {
-			IronSource.Agent.showInterstitial ();
+		if (interstitialAd.IsAdReady()) {
+			interstitialAd.ShowAd();
 		} else {
-			Debug.Log ("unity-script: IronSource.Agent.isInterstitialReady - False");
+			Debug.Log ("unity-script: interstitialAd.IsAdReady - False");
 		}
 	}
 	
-	void InterstitialOnAdReadyEvent(IronSourceAdInfo adInfo)
+	void InterstitialOnAdLoadedEvent(LevelPlayAdInfo adInfo)
 	{
-		Debug.Log("unity-script: I got InterstitialOnAdReadyEvent With AdInfo " + adInfo);
+		Debug.Log("unity-script: I got InterstitialOnAdLoadedEvent With AdInfo " + adInfo);
```

Comment:
same, if we should print string rather than the adInfo object?

Reply:
we don't need `ToString` because we override it already